libafl_frida/src/asan_rt.rs

```diff
@@ -321,8 +327,10 @@ impl AsanRuntime {
     fn hook_realloc(&mut self, ptr: *mut c_void, size: usize) -> *mut c_void {
         unsafe {
             let ret = self.allocator.alloc(size, 0x8);
-            if ptr != std::ptr::null_mut() {
-                (ptr as *mut u8).copy_to(ret as *mut u8, self.allocator.get_usable_size(ptr));
+            if ptr != std::ptr::null_mut() && ret != std::ptr::null_mut() {
```

Comment:
I suggest to use `core` everytime it's avaliable, then it also works for no_std.